openpype/pipeline/context_tools.py

```diff
@@ -658,3 +661,70 @@ def get_process_id():
     if _process_id is None:
         _process_id = str(uuid.uuid4())
     return _process_id
+
+
+def get_current_context_template_data():
+    """Template data for template fill from current context
+
+    Returns:
+        Dict[str, Any] of the following tokens and their values
+        Supported Tokens:
+            - Regular Tokens
+                - app
+                - user
+                - asset
+                - parent
+                - hierarchy
+                - folder[name]
+                - root[work, ...]
+                - studio[code, name]
+                - project[code, name]
+                - task[type, name, short]
+
+            - Context Specific Tokens
+                - assetData[frameStart]
+                - assetData[frameEnd]
+                - assetData[handleStart]
+                - assetData[handleEnd]
+                - assetData[frameStartHandle]
+                - assetData[frameEndHandle]
+                - assetData[resolutionHeight]
+                - assetData[resolutionWidth]
+
+    """
+
+    # pre-prepare get_template_data args
+    current_context = get_current_context()
+    project_name = current_context["project_name"]
+    asset_name = current_context["asset_name"]
+    anatomy = Anatomy(project_name)
+
+    # prepare get_template_data args
+    project_doc = get_project(project_name)
+    asset_doc = get_asset_by_name(project_name, asset_name)
+    task_name = current_context["task_name"]
+    host_name = get_current_host_name()
+
+    # get regular template data
+    template_data = get_template_data(
+        project_doc, asset_doc, task_name, host_name
+    )
+
+    template_data["root"] = anatomy.roots
+
+    # get context specific vars
+    asset_data = asset_doc["data"].copy()
+
+    # compute `frameStartHandle` and `frameEndHandle`
+    if "frameStart" in asset_data and "handleStart" in asset_data:
```

Comment:
Are these data used somewhere? We need to remove the `"assetData"` from here.

Reply:
The point of all template data functions is that they return the same values. This is unique and does not return the same. If those data are needed on other places (other usecase than houdini environments), we can add it to base function `get_template_data`. Otherwise it's maintainance hell...

Reply:
I think this is used for the case where we want to set environment variables pointing to `{frameStartHandle}`/`{frameEndHandle}`

Reply:
Hello, 
Is this sufficient ?  https://github.com/ynput/OpenPype/pull/5846

Reply:
@MustafaJafar I believe @iLLiCiTiT already made a PR actually here https://github.com/ynput/OpenPype/pull/5845